planemo/workflow_lint.py

```diff
@@ -99,6 +102,71 @@ def _lint_tsts(path, lint_context):
             lint_context.valid(f"Tests appear structurally correct for {runnable.path}")
 
 
+def _lint_best_practices(path, lint_context):  # noqa: C901
+    """
+    This function duplicates the checks made by Galaxy's best practices panel:
+    https://github.com/galaxyproject/galaxy/blob/5396bb15fe8cfcf2e89d46c1d061c49b60e2f0b1/client/src/components/Workflow/Editor/Lint.vue
+    """
```

Comment:
It's not ideal for this to duplicate the best practices code in the Galaxy frontend but it's not too much work to maintain, I hope. It seems from #1211 like there is no obvious solution

Reply:
At least for gxformat2 this is (mostly) redundant, can you disable this for gxformat2 workflows:
```
 /tmp  ~/.local/bin/planemo workflow_lint wf14-unlinted-best-practices.yml           ✔  12817  11:04:25  planemo
Applying linter structure... FAIL
.. ERROR: Validation failed - tried _RecordLoader but
  Trying 'GalaxyWorkflow'
wf14-unlinted-best-practices.yml:4:1:       the `inputs` field is not valid because:
                                             - tried _ArrayLoader but
                                               Expected a list
                                             - tried _RecordLoader but
                                               Trying 'WorkflowInputParameter'
wf14-unlinted-best-practices.yml:7:5:             the `position` field is not valid because:
                                                   - tried _PrimitiveLoader but
                                                     Expected a type but got CommentedMap
                                                   - tried _RecordLoader but
                                                     Trying 'StepPosition'
wf14-unlinted-best-practices.yml:8:7:                   invalid field `bottom`, expected one of:
                                                       `top`, `left`
wf14-unlinted-best-practices.yml:20:1:     the `steps` field is not valid because:
                                             tried _ArrayLoader but
                                               - tried _ArrayLoader but
                                                 Expected a list
                                               - tried _RecordLoader but
                                                 Trying 'WorkflowStep'
wf14-unlinted-best-practices.yml:24:5:             the `position` field is not valid because:
                                                     - tried _PrimitiveLoader but
                                                       Expected a type but got CommentedMap
                                                     - tried _RecordLoader but
                                                       Trying 'StepPosition'
wf14-unlinted-best-practices.yml:25:7:                   invalid field `bottom`, expected one of:
                                                         `top`, `left`
- tried _ArrayLoader but
  Expected a list
Applying linter tests... WARNING
.. WARNING: Workflow missing test cases.
```

Reply:
Hmm, some of the best practices are not required though, and we don't have a `recommend` property or something like it in the schema ... never mind, I guess it makes sense